### PR TITLE
Add the `pipeline` subdocument to the `Sources` document

### PR DESCRIPTION
**Pull request by linghengqian**
### Description
_Describe what this change achieves._

- Add the `pipeline` subdocument to the `Sources` document.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

- Fixes https://github.com/opensearch-project/data-prepper/issues/4922  .

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._

- all

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

- No need.

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dlvenable — APPROVED** [review]
Thank you @linghengqian for this contribution!

@kolchfa-aws , Does this look good from you perspective? If so, I'd like to get this in. This functionality is applicable for all versions of Data Prepper.

File: _data-prepper/pipelines/configuration/sources/pipeline.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+layout: default
+title: pipeline 
+parent: Sources
+grand_parent: Pipelines
+nav_order: 55
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/pipeline.md:6]: The topic names under "Sources" are not alphabetized. Could you fix that as part of this PR? Use Pascal numbering (multiples of 10).

File: _data-prepper/pipelines/configuration/sources/pipeline.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/pipeline.md:1]: The `configuration` directory seems unnecessary. Just want to flag this for a future fix. This just makes the link longer (and shorter links are preferable for SEO). We can fix that by removing the `configuration` folder and adding redirects from the old link to the new link in all files under the `pipelines` directory. @vagimeli Could you please put in an issue to fix this?

File: _data-prepper/pipelines/configuration/sources/pipeline.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+layout: default
+title: pipeline 
+parent: Sources
+grand_parent: Pipelines
+nav_order: 55
+---
+
+# pipeline
+
+Use the `pipeline` sink to write to another pipeline.
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/pipeline.md:11]:
```suggestion
Use the `pipeline` sink to read from another pipeline.
```

File: _data-prepper/pipelines/configuration/sources/pipeline.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+## Configuration options
+
+The `pipeline` sink supports the following configuration options.
+
+| Option | Required | Type   | Description                    |
+|:-------|:---------|:-------|:-------------------------------|
+| name   | Yes      | String | Name of the pipe to read from. |
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/pipeline.md:19]:
```suggestion
| `name`   | Yes      | String | The name of the pipeline to read from. |
```

File: _data-prepper/pipelines/configuration/sources/pipeline.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+## Usage
+
+The following example configures a `pipeline` sink that reads from a pipeline named `movies`:
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/pipeline.md:23]:
```suggestion
The following example configures a `pipeline` sink named `sample-pipeline` that reads from a pipeline named `movies`:
```

File: _data-prepper/pipelines/configuration/sources/pipeline.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+```
+sample-pipeline:
+  source:
+    - pipeline:
+        name: "movies"
+```
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/pipeline.md:30]:
````suggestion
```
{% include copy.html %}
````

File: _data-prepper/pipelines/configuration/sources/pipeline.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+| Option | Required | Type   | Description                    |
+|:-------|:---------|:-------|:-------------------------------|
+| name   | Yes      | String | Name of the pipe to read from. |
+
+## Usage
+
+The following example configures a `pipeline` sink that reads from a pipeline named `movies`:
+
+```
```
Comment (kolchfa-aws) [inline _data-prepper/pipelines/configuration/sources/pipeline.md:25]:
```suggestion
```yaml
```

File: _data-prepper/pipelines/configuration/sources/pipeline.md (hunk 6 above)
Comment (kolchfa-aws) [reply _data-prepper/pipelines/configuration/sources/pipeline.md:30]: Looking at the Data Prepper section, none of the yaml snippets have the `yaml` highlighter or copy button added. @vagimeli Could you please put in an issue to fix this?

**Review by kolchfa-aws — CHANGES_REQUESTED** [review]
Thank you, @linghengqian! Just a couple of fixes. I've added a suggestion to fix the topic order as part of this PR but please feel free to put in an issue instead. Thanks!

File: _data-prepper/pipelines/configuration/sources/pipeline.md (hunk 2 above)
Comment (vagimeli) [reply _data-prepper/pipelines/configuration/sources/pipeline.md:1]: Issue https://github.com/opensearch-project/documentation-website/issues/8223

File: _data-prepper/pipelines/configuration/sources/pipeline.md (hunk 6 above)
Comment (vagimeli) [reply _data-prepper/pipelines/configuration/sources/pipeline.md:30]: Issue https://github.com/opensearch-project/documentation-website/issues/8223

File: _data-prepper/pipelines/configuration/sources/pipeline.md (hunk 1 above)
Comment (linghengqian) [reply _data-prepper/pipelines/configuration/sources/pipeline.md]: - I don't seem to see any directories following the so-called `Pascal numbering (multiples of 10)`. I made a new commit, should it look like this?

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @linghengqian! Moving the PR into editorial review. Once it's done, we'll be ready to merge.

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**